### PR TITLE
Added variance filtering to custom/matrixfilter

### PR DESCRIPTION
**Pull request by thomgiles**
Pull request to add filtering on variance for custom/matrixfilter. 

custom/matrixfilter filters a matrix based on various input parms. after slack discussion with Jonathan Manning. 

@pinin4fjords: I have implemented the logic in both the module and test dirs. I've also tested locally but am unsure as to how best to generate the md5s for /tests/modules/nfcore/custom/matrixfilter/test.yml

## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [x] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [x] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [x] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [x] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [x] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/custom/matrixfilter/templates/matrixfilter.R [hunk 1]
```diff
@@ -160,55 +161,67 @@ if ((opt\$sample_file != '') && ( ! is.null(opt\$grouping_variable))){
...
 boolean_matrix <- t(apply(abundance_matrix, 1, function(row) {
-    sapply(tests, function(f) f(row))
+  sapply(tests, function(f) f(row))
 }))
 
-# We will retain features passing all tests
-
-keep <- apply(boolean_matrix, 1, all)
+# Apply the 'most_variant_test' function to identify the most variant rows and add the result to the boolean matrix
+most_variant_vector <- most_variant_test(abundance_matrix)
```
Comment (pinin4fjords) [inline modules/nf-core/custom/matrixfilter/templates/matrixfilter.R:193]: I'd rather you wrapped this in the conditional above. The result should be the same as what you do here, but if I don't configure variance filtering I'd rather this code didn't run at all.

**Review by pinin4fjords — CHANGES_REQUESTED** [review]
Pretty good work!

Could you preserve the existing indents please? 

Nothing new should happen if the variance filtering isn't activated, so I'd rather the new code was within the conditional.

And yes- we need to switch to nf-test. I can help with this, but not for a week or two (work retreat thing).

File: modules/nf-core/custom/matrixfilter/templates/matrixfilter.R (hunk 1 above)
Comment (thomgiles) [reply modules/nf-core/custom/matrixfilter/templates/matrixfilter.R]: Thanks. I've streamlined it to a single conditional block and fixed the whitespace formatting.

**Comment by SPPearce** [comment]
Hi @thomgiles , @pinin4fjords , do you need some help to finish this off?

**Comment by pinin4fjords** [comment]
Thanks once again @SPPearce , this looks fine to me, sorry I forgot about it @thomgiles

**Review by pinin4fjords — APPROVED** [review]

**Comment by pinin4fjords** [comment]
@nf-core-bot fix linting

**Comment by pinin4fjords** [comment]
Thanks @SPPearce ! Sorry, I'd wandered off again

**Pull request merged**